src/Console/Command/ObfuscateData.php

```diff
@@ -0,0 +1,225 @@
+ * @copyright  2018 Ushahidi
+ * @license    https://www.gnu.org/licenses/agpl-3.0.html GNU Affero General Public License Version 3 (AGPL3)
+ */
+
+use Illuminate\Console\Command;
+use Ushahidi\Core\Entity\PostRepository;
+use Ushahidi\Core\Entity\ContactRepository;
+use Ushahidi\Core\Entity\ConfigRepository;
+use Ushahidi\Core\Entity\UserRepository;
+use Ohanzee\DB;
+use Faker;
+
+class ObfuscateData extends Command
+{
+    private $postRepository;
+    private $contactRepository;
+
+    /**
+     * The console command name.
+     *
+     * @var string
+     */
+    protected $name = 'db:obfuscatedata';
+
+    /**
+     * The console command signature.
+     *
+     * @var string
+     */
+    protected $signature = 'db:obfuscatedata';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Obfuscates selected PII and removes various config keys from database';
+
+    public function __construct()
+    {
+        parent::__construct();
+    }
+
+    public function handle()
+    {
+        $this->db = service('kohana.db');
+        $this->contactRepository = service('repository.contact');
+        $this->postRepository = service('repository.post');
+        $this->configRepository = service('repository.config');
+        $this->userRepository = service('repository.user');
+
+        if ($this->isTestDeployment() || $this->isStagingDeployment()) {
+            //confirm acknowledgement
+            $this->info("This script will wipe data from the current deployment.\n\n");
+
+            if (!$this->confirm('Do you acknowledge that this data will still contain sensitive data?')) {
+                $this->info("Request canceled.\n");
+                exit;
+            }
+
+            // Do overwriting
+            $this->db->begin();
+            $this->overwriteContacts();
+            $this->overwritePostAuthors();
+            $this->removeDataProviderValues();
+            $this->overwriteSiteConfig();
+            $this->deleteUsers();
+            $this->addAdminUser();
+            $this->db->commit();
+        } else {
+            $this->info("This script will only run on test or staging deployments.\n\n");
+            exit;
+        }
+    }
+
+    private function isTestDeployment()
+    {
+        return (strtolower(getenv('APP_ENV')) == 'test');
+    }
+
+    private function isStagingDeployment()
+    {
+        return (strtolower(getenv('APP_ENV')) == 'staging');
+    }
+
+    /* optional implementations...
+    private function randomizeLongitudeLatitude($boundaries)
+    {
+
+    }
+    */
+
+    /* TODO: would this be useful?
+    private function scrambleLongitudeLatitude($distanceFromOrigin = null)
+    {
+
+    }
+    */
+
+    private function overwriteContacts()
+    {
+        $this->info("Overwriting contact data...");
+        $faker = Faker\Factory::create();
+        $query = DB::select('contacts.id', 'contacts.type', 'contacts.contact')
```

Comment:
Do you think this should all move into contacts repo? or stay here as a read/write step so that ContactRepo doesn't have to depend on Faker?
I can't actually decide which makes more sense

Reply:
I'm not sure either. I suppose each of these repos could have an obfuscation method, and possibly even a fake data generation method.